docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,191 @@
+
+The output of the aggregation script should be loaded to some existing visualization tool, like grafana.
+
+It may be nice to host raw and aggregate data on separate servers - then even if the data presented on the dashboard is compromised, the raw data will be not exposed to the world.
+
+Having both server and aggregation scripts in Python will reduce maintenance costs for napari core devs.
+
+We should register the `telemetry.napari.org` domain and use it for the server. The main page will contain this NAP and a link to the summary dashboard.
+
+
+The main part of the application side should be implemented in `napari-telemetry` package. 
+The package should not report in stream mode, but collect data on the disk and send it in batches. This will reduce the risk of leaking data. The package should implement a utility to allow users to preview collected data before sending it to the server.
+
+In napari itself, the following changes should be implemented:
+
+1) The indicator that shows the telemetry status
+2) The dialog that asks a user if they want to enable telemetry
+3) code to check if telemetry is enabled (to not load the `napari-telemetry` package if it is disabled)
+4) code required to init `napari-telemetry` package
+
+
+## Potential problems 
+
+There is a risk that someone may try to highjack the telemetry module name to have code executed at every napari start. 
+
+I do not expect that it is a high risk, but exists. We could address it by code signing. This will require additional procedures to protect private cryptographic keys.
+
+Another option is to scan public plugins and their dependencies. This is simpler but will require establishing additional communication channels to be able to warn users about the potential problem. 
+
+
+## GDPR compliance
+
+I'm almost sure that we will not collect data that are covered by GDPR. But to get better atmosphere 
+we need to add instruction how user could retrive his unique identifier and setup a process 
+for requests to remove data from the server. It is not high propability of usage as life span of data is short,
+but we need to be prepared for such a situation. I suggest to use e-mail for that.
+
+
+  
+## Backward Compatibility 
+  
+ Not relevant
+  
+## Future Work 
+
+A nice extension may be the ability for the steering council to create a certificate of telemetry output that could be given to plugin maintainers to prove to supervisors that their plugin is used by the community. 
+
+  
+ ## Alternatives 
+
+ During the discussion, there is a proposal to use the same approach as used in ImageJ. 
+ 
+ Mean that instead of implementing telemetry on the client side we could implement it on the update server side. The advantage and disadvantage of such a solution is that no user could opt out of telemetry. Also, such a method could potentially provide information about the Python version, napari version and list of installed plugins. All others will require a mechanism from this NAP.
```

Comment:
```suggestion
This would mean instead of implementing telemetry on the client side we could implement it on the update server side. The advantage and disadvantage of such a solution is that no user could opt out of telemetry. Also, such a method could potentially provide information about the Python version, napari version and list of installed plugins. All others will require a mechanism from this NAP.
```